server.js

```
@@ -22,9 +22,44 @@ server.use({
 		.catch(function() { next() });
 	}
 });
+server.use({
+  // Prerender does not propagate headers, which is extremely annoying for crawlers
+  // This takes all headers sent to prerender, and puts them in the PhantomJS request
+  // Although it was just a few LoC, it took a lot of blood, frust, and keyboards to build
+  // Treat it accordingly.
+  onPhantomPageCreate: function (phantom, req, res, next) {
+    req.prerender.page.run(req.headers, function (headers, resolve) {
+      var customHeaders = this.customHeaders || {};
+      for (var header in headers) {
+        if (headers.hasOwnProperty(header)) {
+          customHeaders[header] = headers[header];
+        }
+      }
+      this.customHeaders = customHeaders;
+
+      resolve();
+    }).then(function () {
+      next();
+    }).catch(function () {
+      console.err('Error while propagating headers');
+      next();
```

Shouldn't the error also be propagated? Now the error is just swallowed, which may result in unexpected behaviour.